services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/ServiceMessageCodec.java

```diff
@@ -43,11 +131,11 @@ public ServiceMessageCodec(HeadersCodec headersCodec) {
     } else if (message.hasData()) {
       dataBuffer = ByteBufAllocator.DEFAULT.buffer();
       try {
-        DataCodec dataCodec = DataCodec.getInstance(message.dataFormatOrDefault());
+        DataCodec dataCodec = getDataCodecByContentType(message.dataFormatOrDefault());
         dataCodec.encode(new ByteBufOutputStream(dataBuffer), message.data());
       } catch (Throwable ex) {
         ReferenceCountUtil.safestRelease(dataBuffer);
-        LOGGER.error("Failed to encode data on: {}, cause: {}", message, ex.toString());
+        LOGGER.error("Failed to encode data on: {}, cause: {}", message, ex);
```

Comment:
Will it print empty curly braces (`cause: {}`)?

Reply:
I put similar comment elsewhere in this comment.